### PR TITLE
Drop older versions

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is for the next major version.

## Changelog

```markdown
### Removed
- Remove support for Symfony ^4.4 and 5.3.x
- Remove support for Twig ^2
- Remove support for `doctrine/common` ^2
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
This is what I would like to do, but not sure if this is the correct place:

1. Drop old versions of doctrine/common and twig (I think this is the correct place, we have done like this on other majors)
2. Drop older sf versions (this can probably be done on 4.x, but is the correct move or should we wait for 5.0?)

**Comment by VincentLanglet** [comment]
I would drop only on 5.x IMHO.

**Comment by jordisala1991** [comment]
So, you agree on all the changes proposed here?

**Comment by VincentLanglet** [comment]
> So, you agree on all the changes proposed here?

I'm in favor of dropping support for symfony 4.4.

For twig and doctrine/common, I don't know if we win anything by dropping the support.
Also, I dunno if some user could have a real reason to be stuck with doctrine common 2 or twig 2.
So, I would say I'm ok with the drop, but don't know the impact.

**Comment by jordisala1991** [comment]
Dev-kit PR: https://github.com/sonata-project/dev-kit/pull/2014

File: phpstan.neon.dist [hunk 1]
```diff
@@ -14,3 +14,5 @@ parameters:
     checkMissingVarTagTypehint: true
     checkMissingTypehints: true
     checkUninitializedProperties: true
+    scanFiles:
+        - vendor/symfony/dependency-injection/Loader/Configurator/ContainerConfigurator.php
```
Comment (VincentLanglet) [inline phpstan.neon.dist:18]: Why is this needed ? Vendor is generally always scanned isn't it ?

Comment (jordisala1991) [reply phpstan.neon.dist:18]: https://github.com/phpstan/phpstan-symfony#configuration

If I don't add this I can't use `service`, `param` or `abstract_arg` since phpstan reports them as not existent symbols.

**Comment by jordisala1991** [comment]
Should we also drop php7? (of course in another PR)

**Comment by VincentLanglet** [comment]
> Should we also drop php7? (of course in another PR)

I think we can.

**Review by VincentLanglet — APPROVED** [review]
